web/source/kmwutils.ts

```diff
@@ -51,6 +51,14 @@ namespace com.keyman {
       this.keyman = keyman;
     }
 
+    ['getBCP47Components'](lgCode: string): string[] {
+      if(lgCode.indexOf('-')==-1) {
+        return [lgCode];
+      } else {
+        return lgCode.split('-');
+      }
+    }
+
```

Comment:
This is fine for getting the language code portion of the BCP 47 tag. (Rename to `getLanguageCode`?)

Later on if we need to parse out the script, region, variant components, maybe we should see what's available on npm.
(Off a quick search, maybe [language-tags](https://www.npmjs.com/package/language-tags)?)